src/main_service/main_service.rs

```diff
@@ -106,44 +145,47 @@ impl Service for MainService {
     fn run(&mut self) -> Result<ServiceHandles<()>, String> {
         info!("{}.run | Starting...", self.dbg_id);
         let dbg_id = self.dbg_id.clone();
+        let conf_tree = self.conf.clone();
         let exit = self.exit.clone();
         info!("{}.run | Preparing thread...", dbg_id);
+
         let handle = thread::Builder::new().name(format!("{}.run", dbg_id.clone())).spawn(move || {
             let interval = 0.0; // from sampl_freq & buf_size
             let interval = Duration::from_secs_f64(interval);
             let mut cycle = ServiceCycle::new(&dbg_id, interval);
             let addr = "127.0.0.1:15181"; //other address in string format
 
-            let parent = "";
-            let conf_tree = ConfTree::new();
-            let config = MainService::new(parent, conf_tree);
-            for(freq, amp, phi) in config.signal{
-                let amplitude = amp;
-                let angle = phi;
-            }
-            let buf:[u8; 0] = [];
-            //let header = todo!("Udp message head, find detales here: https://github.com/a-givertzman/cma-server/issues/123#issue-2478437558");
+            let config = MainService::new(&dbg_id, conf_tree);
+            let mut buf = Vec::with_capacity(512);
+
             loop {
-                match Self::udp_bind(addr) {
-                    Ok(socket) => {
-                        cycle.start();
-                        // 
-                        // do the sampling actions
-                        // - calc amplitude
-                        // - add to buffer
-                        // - if buffer is full
-                        //      do the udp actions
-                        //      - build data splitting buffer amplitudes into bytes
-                        //      - build message using header and data
-                        //      - send message to the socket
-                        let bytes = buf.map(|amp: f64| {
-                            let amp: i16 = amp.round() as i16;
-                            let amp_bytes = amp.to_be_bytes();
-                        });
-                        socket.send(bytes);
-                        cycle.wait();
+
+                for (freq, amp, phi) in config.conf.signal.iter(){
+                    let amplitude = amp;
+                    buf.push(amplitude);
+
+                    if (buf.capacity() == buf.len()){
```

Comment:
I thought `src/buffer.rs` - was implemented exactly for accumulating with specified capacity like:
```rust
match buf.add(amplitude) {
   Some() => {
      ...
   }
   None -> {
      ...
   }
}
```